### DISCUSSION TITLE
workspace or files from API or URL

### DISCUSSION THREAD

**Discussion by julen9** [eclipse-emfcloud/emfcloud#222] [Coffee Editor]
Hi everyone,

I am working with coffee-editor project. I have edited the model server adding new elements on it. Now, my question is:

is possible to load and save files (or a complete workspace) from an API or URL? I need to give to the users the IDE Eclipse Theia but with my own workspace with my files (hosted in a DB but served throught an API). The users can edit, add and delete files into the workspace and save it remotely (maybe interacting with the API)

Is it possible?
Thanks a lot in advance,

Regards,
Julen.

**Comment by eneufeld** [comment]
Hi @julen9 ,

sorry for the late reply.
The current implementation is based on the workspace. If you don't want to change the code of the modelserver the simplest thing to do would add a custom theia extension which would interact with your API and basically clone the data into the workspace.
As far as I understand this is very similar to a git flow: with clone, commit, push .

Best,
Eugen
